boards/nrf9160dk/Kconfig

```
@@ -0,0 +1,17 @@
+# Copyright (c) 2021 Mesotic SAS
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config BOARD
+    default "nrf9160dk" if BOARD_NRF9160DK
+
+config BOARD_NRF9160DK
+    bool
+    default y
+    select CPU_MODEL_NRF9160
+    select HAS_PERIPH_TIMER
+    select HAS_PERIPH_TIMER_PERIODIC
+    select HAS_PERIPH_UART
+    select HAS_PERIPH_UART_MODECFG
```

```suggestion
    select HAS_PERIPH_UART
```

That was correct, those are provided by the CPU

---

We don't inherit Kconfig from nrf5x for now as this port doesn't support all features. This may be changed later when we add all missing features but I think we should leave as it for now.

---

Better move it to the CPU Kconfig still so it's all in the right place already.

---

I moved `HAS_PERIPH_TIMER_PERIODIC` and `HAS_PERIPH_UART_MODECFG` to CPU level.

I checked a bit how other boards declare `HAS_PERIPH_UART` and `HAS_PERIPH_TIMER`. These features are declare at board level probably because it depends on the configuration from `periph_conf.h` so I let them in the board Kconfig.